app/javascript/onboarding/components/ProfileForm.jsx

```diff
@@ -0,0 +1,132 @@
+import { h, Component } from 'preact';
+import PropTypes from 'prop-types';
+
+import Navigation from './Navigation';
+import { userData, getContentOfToken, updateOnboarding } from '../utilities';
+
+class ProfileForm extends Component {
+  constructor(props) {
+    super(props);
+
+    this.handleChange = this.handleChange.bind(this);
+    this.onSubmit = this.onSubmit.bind(this);
+    this.user = userData();
```

Comment:
@vaidehijoshi  
> Also, what happens if the userData() isn't loaded and this tries to render? 

It’s safeguarded in `app/javascript/packs/Onboarding.jsx`. If `getUserDataAndCsrfToken()` fails, onboarding won’t happen.